src/user/user.controller.ts

```diff
@@ -22,8 +24,9 @@ export class UserController {
 
   @UseGuards(JwtAuthGuard)
   @Get()
-  async getAll() {
-    return await this.service.findAll();
+  async getAll(@Query() query) {
+    const queryParams = new QueryParams(query);
+    return await this.service.findAll(queryParams);
```

Comment:
Не забывай про аннотации. В данном случае это `Record<string, string>`. Но нам нужно точно знать какие могут быть ключи: `Record<queryParamKey, string>`. Надо создать какой-нибудь enum для возможных ключей на проекте

Reply:
```suggestion
  async getAll(@Query() query: Record<ParamsList, string>) {
    const queryParams = new QueryParamsList(query);
    return await this.service.findAll(queryParams);
```
Импортируемые ключи:
```
export type ParamsList = keyof typeof enumParamsList;
export type ParamsSingle = keyof typeof enumParamsSingle;
```